src/test/java/uk/gov/companieshouse/filinghistory/api/repository/RepositoryTest.java

```diff
@@ -44,42 +44,62 @@ class RepositoryTest {
     private MongoTemplate mongoTemplate;
 
     @Mock
-    private AggregationResults<FilingHistoryListAggregate> aggregationResults;
+    private AggregationResults<FilingHistoryIds> aggregationResultsFilingHistoryIds;
     @Mock
-    private FilingHistoryListAggregate listAggregate;
+    private AggregationResults<FilingHistoryDocument> aggregationResultsFullDocuments;
+    @Mock
+    private FilingHistoryIds filingHistoryIds;
+    @Mock
+    private List<FilingHistoryDocument> mockFilingHistoryDocuments;
     @Mock
     private AggregationResults<FilingHistoryDeleteAggregate> deleteAggregationResults;
     @Mock
     private FilingHistoryDeleteAggregate deleteAggregate;
 
     @Test
-    void shouldCallMongoTemplateWithCompanyNumberCriteriaOnly() {
+    void shouldCallMongoTemplateSuccessfullyWithCompanyNumberCriteriaOnly() {
```

Comment:
These tests are testing too much. There should be one repository method called per test which tests a specific part of that method. Single responsibility should apply to source code and unit tests.